src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2154,9 +2154,22 @@ private ISwitchExpressionOperation CreateBoundSwitchExpressionOperation(BoundSwi
         {
             IOperation value = Create(boundSwitchExpression.Expression);
             ImmutableArray<ISwitchExpressionArmOperation> arms = CreateFromArray<BoundSwitchExpressionArm, ISwitchExpressionArmOperation>(boundSwitchExpression.SwitchArms);
+
+            bool isExhaustive = !boundSwitchExpression.ReportedNotExhaustive;
```

Comment:
I don't think we should be doing any of the other logic here. It should just be this.

Reply:
Even if this does compensate for a different bug, we shouldn't really do that. We should just fix the other bug and leave this code correct.

Reply:
Yes, that is what I would prefer, too.
But the result should be always the same regardless of nullable enabled or disabled.

Reply:
>  But the result should be always the same regardless of nullable enabled or disabled.

I don't agree. The compiler intentionally does not warn if the user did not cover the `null` case when `#nullable enable` is specified and the code said `null` was not valid. The switch expression _is_ exhaustive.

Reply:
And this is what should be answered by this property

Reply:
> But nevertheless the compiler generates an arm which throws a SwitchExpressionException - even if no warning was emitted / issued - or if they were filtered out.

I don't really see why that matters: the question this API is answering is whether the given switch expression arms were exhaustive. They are.

Reply:
If I would be only interested in warnings I could just call "GetDiagnostics()"

Reply:
This is a question about the semantics of the code. C# considers the switch expression to be fully exhaustive. Whether a warning or error is emitted (and remember, warnings can be turned off by users, so `GetDiagnostics()` would not work) doesn't really matter here.

Reply:
I mean that this flag should be set before nullable warnings get filtered out.

Reply:
To make it short: IsExhaustive should return `true` when there is not artificial switch-expression-arm generated and otherwise `false`.
Do you agree?

Reply:
It should return true when the language considers the switch expression to be exhaustive.

Reply:
Alright, @bernd5 we discussed this in the compiler team meeting this afternoon, and came to the conclusion that we should make this API return false if the compiler generates a default branch, regardless of the current nullable state. The appropriate way to do this is to add a new field to the bound tree that tracks whether a default label is present, and record that information when the switch expression is being bound. https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Binder/SwitchExpressionBinder.cs#L73 this function will want to return two bools, one for whether a warning was reported, and one for whether the default label was generated, which the operation factory will look at. 

Separately, we need to either be very specific about the doc comment wording here to avoid confusion, and potentially change the name of the property. My suggestion would be something like `HasImplicitThrow`, but @RikkiGibson or @AlekseyTs might have other suggestions (including keeping the existing name).

Reply:
I'd prefer keeping the existing name - but `HasImplicitThrow` is okay for me, too.

Reply:
My main concern about `IsExhaustive` is that it's too wrapped up in whether a warning about exhaustiveness was given.

Reply:
> Separately, we need to either be very specific about the doc comment wording here to avoid confusion, and potentially change the name of the property. 

I think the current name is fine. Adding a clear doc comment should be sufficient to avoid a confusion.